rust/worker/src/storage/s3.rs

```diff
@@ -155,56 +158,177 @@ impl S3Storage {
         }
     }
 
-    pub(crate) async fn put_bytes(&self, key: &str, bytes: Vec<u8>) -> Result<(), S3PutError> {
-        let bytestream = ByteStream::from(bytes);
-        self.put_bytestream(key, bytestream).await
-    }
-
-    pub(crate) async fn put_file(&self, key: &str, path: &str) -> Result<(), S3PutError> {
-        let bytestream = ByteStream::from_path(path).await;
-        match bytestream {
-            Ok(bytestream) => return self.put_bytestream(key, bytestream).await,
-            Err(e) => {
-                return Err(S3PutError::S3PutError(e.to_string()));
+    pub(crate) async fn put_bytes(
+        &self,
+        key: &str,
+        bytes: Vec<u8>,
+        part_size_bytes: Option<u64>,
```

Comment:
I would prefer if this was just taken as config as mentioned below and you couldn't override it for now, to minimize the complexity of this interface. If in the future we need customizable part size then we can extend it